### PR TITLE
Feature/3150 idea hub module scaffold

### PR DESCRIPTION
**Pull request by ivankruchkoff**
## Summary

Addresses issue #3150

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: tests/phpunit/integration/Modules/Idea_HubTest.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+use Google\Site_Kit\Tests\TestCase;
+
+/**
+ * @group Modules
+ */
+class Idea_HubTest extends TestCase {
+	use Module_With_Settings_ContractTests;
+
+	public function test_register() {
+		$analytics = new Idea_Hub( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
```
Comment (aaemnnosttv) [inline tests/phpunit/integration/Modules/Idea_HubTest.php:29]: Please make sure variables are updated accordingly 😄

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @ivankruchkoff ! This looks good for the most part, just a few small things to address.

File: includes/Modules/Idea_Hub.php [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	 * @return array Map of datapoints to their definitions.
+	 */
+	protected function get_datapoint_definitions() {
+		return array(
+			'GET:draft-post-ideas'        => array( 'service' => '' ),
+			'GET:new-ideas'               => array( 'service' => '' ),
+			'GET:published-post-ideas'    => array( 'service' => '' ),
+			'GET:saved-ideas'             => array( 'service' => '' ),
+			'POST:create-idea-draft-post' => array( 'service' => '' ),
+			'POST:update-idea-state'      => array( 'service' => '' ),
```
[the remark this answers is not in the document]
Comment (ivankruchkoff) [reply includes/Modules/Idea_Hub.php]: Updated in https://github.com/google/site-kit-wp/pull/3223/commits/1fa32b4dce84138e83d87ae3c1b08bff9c5c5207

File: includes/Modules/Idea_Hub.php [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+	protected function setup_info() {
+		return array(
+			'slug'        => self::MODULE_SLUG,
+			'name'        => _x( 'Idea Hub', 'Service name', 'google-site-kit' ),
+			'description' => __( 'TODO.', 'google-site-kit' ),
+			'cta'         => __( 'TODO.', 'google-site-kit' ),
+			'order'       => 7,
+			'homepage'    => __( 'https://www.google.com', 'google-site-kit' ),
+			'learn_more'  => __( 'https://www.google.com', 'google-site-kit' ),
+			'internal'    => true,
```
[the remark this answers is not in the document]
Comment (ivankruchkoff) [reply includes/Modules/Idea_Hub.php]: Updated in https://github.com/google/site-kit-wp/pull/3223/commits/90573808bcd5d276a8abd78f32c17cb695e0671d

File: tests/phpunit/integration/Modules/Idea_HubTest.php (hunk 1 above)
Comment (ivankruchkoff) [reply tests/phpunit/integration/Modules/Idea_HubTest.php]: Updated in https://github.com/google/site-kit-wp/pull/3223/commits/3f6a2d3795733ef1ad792ca5bcd4f0cabec88308

File: includes/Modules/Idea_Hub.php [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	 * @since n.e.x.t
+	 *
+	 * @return array Associative array of module info.
+	 */
+	protected function setup_info() {
+		return array(
+			'slug'        => self::MODULE_SLUG,
+			'name'        => _x( 'Idea Hub', 'Service name', 'google-site-kit' ),
+			'description' => __( 'TODO.', 'google-site-kit' ),
+			'cta'         => __( 'TODO.', 'google-site-kit' ),
```
Comment (aaemnnosttv) [inline includes/Modules/Idea_Hub.php:195]: These are just placeholders and don't need to be translated yet. We wouldn't want this to be something that translators would actually translate either.
```suggestion
			'description' => 'TODO.',
			'cta'         => 'TODO.',
```

File: includes/Modules/Idea_Hub.php [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+			'order'       => 7,
+			'homepage'    => __( 'https://www.google.com', 'google-site-kit' ),
+			'learn_more'  => __( 'https://www.google.com', 'google-site-kit' ),
```
Comment (aaemnnosttv) [inline includes/Modules/Idea_Hub.php:198]: Same here.
```suggestion
			'homepage'    => 'https://www.google.com',
			'learn_more'  => 'https://www.google.com',
```

**Review by aaemnnosttv — COMMENTED** [review]
Looks great, just one last thing regarding the placeholder info for the module. I'll apply the changes now then this should be good to go.

**Review by aaemnnosttv — APPROVED** [review]

**Pull request merged**